plugins/services/src/js/containers/services/ServicesContainer.js

```diff
@@ -432,28 +432,54 @@ class ServicesContainer extends React.Component {
     };
   }
 
-  getModals(service) {
-    const modalProps = Object.assign({}, this.state.modal);
+  /**
+   * This function cleans and updates the modalProps property
+   *
+   * This is needed to refresh the state of the service from the store once the
+   * modal is loaded. In our delete group modal for example we need feedback
+   * of the service while it is being deleted. In other places (after deleting
+   * an app) on the other hand, it is possible that the modal is still open,
+   * DCOSStore returns undefined but we still need the (old) information stored
+   * in modalProps.service - and want to close the modal.
+   *
+   * @param {object} stateService - an object which contains id and a service
+   * @param {ServiceTree} [service] - service information
+   * @returns {object} updated and cleaned up modal information (props)
+   */
+  getUpdatedModalProps(stateService, service) {
+    const modalProps = Object.assign({}, stateService);
 
-    // This is needed to refresh the state of the service from the store once the
-    // modal is loaded. In our delete group modal for example we need feedback
-    // of the service while it is being deleted
-    if (!modalProps.service) {
+    // modalProps.service only is set, when a modal is already open, if we want to
+    // open one, we have to assign the given `service` to modalProps.service
+    if (!modalProps.service && service) {
       modalProps.service = service;
-    } else {
+    }
+
+    // this block updates the information - if possible - or deletes the
+    // modal id from the object - if necessary.
+    if (
+      modalProps.service &&
+      DCOSStore.serviceTree.findItemById(modalProps.service.id)
+    ) {
       modalProps.service = DCOSStore.serviceTree.findItemById(
         modalProps.service.id
       );
+    } else if (modalProps.id) {
+      delete modalProps.id;
```

Comment:
This case looks to always return `{}`, so I think you should just `return {}` here.

Reply:
tried this before, but it breaks some modals :(

Reply:
let me add a bit more context: when `id` is unset, this basicly says "close the modal", since the modal is still open in the next render-interval, the `service` has to be still valid, also unsetting `service` here would break a still-open-but-to-be-closed modal…

Reply:
Ok makes sense.